javascript/ql/src/semmle/javascript/PackageExports.qll

```diff
@@ -70,6 +70,61 @@ private DataFlow::Node getAValueExportedByPackage() {
     result = cla.getAStaticMethod() or
     result = cla.getConstructor()
   )
+  or
+  // *****
+  // Various standard library methods for transforming exported objects.
+  // *****
+  //
+  // Object.defineProperties
+  exists(DataFlow::MethodCallNode call |
+    call = DataFlow::globalVarRef("Object").getAMethodCall("defineProperties") and
+    [call, call.getArgument(0)] = getAValueExportedByPackage() and
+    result = call.getArgument(any(int i | i > 0))
+  )
+  or
+  // Object.defineProperty
+  exists(DataFlow::MethodCallNode call |
+    call = DataFlow::globalVarRef("Object").getAMethodCall("defineProperty") and
+    [call, call.getArgument(0)] = getAValueExportedByPackage()
+  |
+    result = call.getArgument(2).getALocalSource().getAPropertyReference("value")
+    or
+    result =
+      call.getArgument(2)
+          .getALocalSource()
+          .getAPropertyReference("get")
+          .(DataFlow::FunctionNode)
+          .getAReturn()
+  )
+  or
+  // Object.assign
+  exists(DataFlow::MethodCallNode assign |
+    assign = DataFlow::globalVarRef("Object").getAMethodCall("assign")
+  |
+    getAValueExportedByPackage() = [assign, assign.getArgument(0)] and
+    result = assign.getAnArgument()
+  )
+  or
+  // Array.prototype.{map, reduce, entries, values}
+  exists(DataFlow::MethodCallNode map |
+    map.getMethodName() = ["map", "reduce", "entries", "values"] and
+    map = getAValueExportedByPackage()
+  |
+    result = map.getArgument(0).getABoundFunctionValue(_).getAReturn()
+    or
+    // assuming that the receiver of the call is somehow exported
+    result = map.getReceiver()
+  )
+  or
+  // Object.{fromEntries, freeze, entries, values}
+  exists(DataFlow::MethodCallNode freeze |
+    freeze =
+      DataFlow::globalVarRef("Object")
+          .getAMethodCall(["fromEntries", "freeze", "entries", "values"])
```

Comment:
We should also have `freezes` friend `seal` here.

(see https://github.com/github/codeql/blob/746e9948b07f730c8c0a89e6ebcac5fa300b9d30/javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll#L447)